packages/x-components/src/components/highlight.vue

```diff
@@ -0,0 +1,217 @@
+        <span
+          v-if="matchParts.end"
+          v-text="matchParts.end"
+          class="x-highlight__text"
+          data-test="highlight-end"
+        />
+      </span>
+    </slot>
+  </NoElement>
+</template>
+
+<script lang="ts">
+  import { Component, Mixins, Prop } from 'vue-property-decorator';
+  import { normalizeString } from '../utils/normalize';
+  import { VueCSSClasses } from '../utils/types';
+  import { NoElement } from './no-element';
+  import { dynamicPropsMixin } from './dynamic-props.mixin';
+
+  /**
+   * Highlights the given part of the text. The component is smart enough to do matches
+   * between special characters like tilde, cedilla, eñe, capital letters...
+   *
+   * @public
+   */
+  @Component({
+    components: { NoElement }
+  })
+  export default class Highlight extends Mixins(
+    dynamicPropsMixin(['noMatchClass', 'matchingPartClass'])
+  ) {
+    /**
+     * The text to highlight some part of it.
+     *
+     * @public
+     */
+    @Prop({ default: '' })
+    public text!: string;
+
+    /**
+     * The part of the text to be highlighted.
+     *
+     * @public
+     */
+    @Prop({ default: '' })
+    public highlight!: string;
+
+    /**
+     * CSS Class to add when the `text` string contains the `highlight` string.
+     */
+    @Prop({ default: 'x-highlight-text' })
+    public matchClass!: string;
+
+    /**
+     * Checks if the normalized suggestion query matches with the module's query, so it has a
+     * matching part.
+     *
+     * @returns True if there is a match between the text and the highlight strings.
+     * @internal
+     */
+    protected get hasMatch(): boolean {
+      return !!this.matchParts.match;
+    }
+
+    /**
+     * CSS classes to add depending on the component state.
+     *
+     * @remarks
+     * `x-highlight--has-match`: When there is a match between the text and the part to highlight.
+     * @returns The {@link VueCSSClasses} classes.
+     * @public
+     */
+    protected get dynamicCSSClasses(): VueCSSClasses {
+      const classes: VueCSSClasses = {
+        'x-highlight--has-match': this.hasMatch,
+        [this.matchClass]: this.hasMatch
+      };
+      if (this.noMatchClass) {
+        classes[this.noMatchClass] = !this.hasMatch;
+      }
+      return classes;
+    }
+
+    /**
+     * Splits the text to highlight into 3 parts: a starting part, the matching part
+     * and the ending part. If there is no match between the text and the highlight, the `start`
+     * property will contain the whole text.
+     *
+     * @returns An object containing the different parts of the text.
+     * @internal
+     */
+    protected get matchParts(): HighlightMatch {
+      const matcherIndex = normalizeString(this.text).indexOf(normalizeString(this.highlight));
+      return matcherIndex !== -1 && this.highlight
+        ? this.splitAt(this.text, matcherIndex, matcherIndex + this.highlight.length)
+        : { start: this.text, match: '', end: '' };
```

Comment:
The `matcherIndex` is based on the normalized strings of `text` and `highlight` but the `text` and `highlight` length passed to `splitAt` are not normalized so I guess it could return unexpected results in case the original variables contain spaces at start/end or diacritical marks causing the length to change.

Reply:
Good catch. I think by adding a `.trim()` call we are good to go. Combining diacritical marks shouldn't make the text length to change.